internal/mode/static/nginx/config/stream_servers.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"fmt"
+	gotemplate "text/template"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/config/stream"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/dataplane"
+)
+
+var streamServersTemplate = gotemplate.Must(gotemplate.New("streamServers").Parse(streamServersTemplateText))
+
+func executeStreamServers(conf dataplane.Configuration) []executeResult {
+	streamServers := createStreamServers(conf)
+
+	streamServerResult := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(streamServersTemplate, streamServers),
+	}
+
+	result := []executeResult{
+		streamServerResult,
+	}
+
+	return result
+}
+
+func createStreamServers(conf dataplane.Configuration) []stream.Server {
+	if len(conf.TLSPassthroughServers) == 0 {
+		return nil
+	}
+
+	streamServers := make([]stream.Server, 0, len(conf.TLSPassthroughServers)*2)
+	portSet := make(map[int32]struct{})
+	upstreamNameSet := make(map[string]struct{})
+
+	for _, u := range conf.StreamUpstreams {
```

Comment:
Same comment here about just setting UpstreamName to `""` in dataplane.Configuration.